src/commands/import-resolver.ts

```diff
@@ -34,12 +34,12 @@ function resolveSourceSpecifier(spec: string, stats: Stats | false, isBundled: b
   } else if (!stats && !spec.endsWith('.js')) {
     spec = spec + '.js';
   }
-  const ext = path.extname(spec).substr(1);
-  const extToReplace = srcFileExtensionMapping[ext];
+  const {baseExt} = getExt(spec);
+  const extToReplace = srcFileExtensionMapping[baseExt];
   if (extToReplace) {
-    spec = spec.replace(new RegExp(`${ext}$`), extToReplace);
+    spec = spec.replace(new RegExp(`\\${baseExt}$`), extToReplace);
```

Comment:
Here (and in other places), we’re now searching for `\.js$` to replace, instead of `js$`. It probably won’t make a practical difference, but I wanted to make sure that we were matching dots, too.